### ISSUE TITLE
LoggingClient should log headers that are actually sent

### ISSUE THREAD

**Issue by anuraaga** [line/armeria#1941]
Currently, at a very late stage when converting to a Netty object we do some header manipulation, such as scrubbing HTTP/2 pseudoheaders when sending as HTTP/1. This makes the logs confusing and they should log the actually sent headers instead.

**Comment by trustin** [comment]
What's interesting is some users actually want the opposite. They want to know what headers they tried to send. Should we keep both?

**Comment by anuraaga** [comment]
Yeah probably should have both, or an option for both, in that case. We have a couple different type of loggers and I wonder if that can be the knob too.

`LoggingClient/Service` -> What user meant to send / receive, semantically
`AccessLoggingService` and a new `Client` -> What was actually sent / recieved, over the wire
